### PR TITLE
Update Unity bindings

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Update bindings in Unity SDK to consume Apple SDK with updated naming

## Related PRs or issues

[AB#83151](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/83151)

### CODE REVIEW

File: Assets/AppCenter/Plugins/iOS/Core/AppCenterStarter.m [hunk 1]
```diff
@@ -1,3 +1,150 @@
...
+enum StartupMode {
+  APPCENTER,
+  ONECOLLECTOR,
+  BOTH,
+  NONE,
+  SKIP
+};
+
 @implementation AppCenterStarter
-//This file is a stub! We're going to replace it at build.
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/iOS/Core/AppCenterStarter.m]: Revert this

File: build.cake [hunk 2]
```diff
@@ -16,7 +16,7 @@ using System.Threading;
 
 // Native SDK versions
 const string AndroidSdkVersion = "3.3.1";
-const string IosSdkVersion = "3.3.4";
+const string IosSdkVersion = "4.0.0+a17972050";
```
Comment (MatkovIvan) [inline build.cake]: I guess this is invalid number/url

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
 # App Center SDK for Unity Change Log
 
+## Release 3.3.2 (Under development)
+
+### App Center Analytics
+
+#### iOS
+
+[Fix] Fix naming conflict with iOS 14 private Apple framework.
```
Comment (annakocheshkova) [inline CHANGELOG.md]: Should we copy the message from the Apple changelog?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,15 @@
 # App Center SDK for Unity Change Log
 
+## Release 3.3.2 (Under development)
+
+### App Center Analytics
```
Comment (annakocheshkova) [inline CHANGELOG.md]: App Center Analytics only?

**Review by annakocheshkova — DISMISSED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (AnastasiaKubova) [reply CHANGELOG.md:5]: Yes, conflict names occurred only with App Center Analytics module

**Review by MatkovIvan — APPROVED** [review]

**Review by olmisa — APPROVED** [review]

**Review by vvechkanov — APPROVED** [review]

**Pull request merged**